### ISSUE TITLE
Display/query default config even when not present in config file

### ISSUE THREAD

**Issue by poliorcetics** [nushell/nushell#5436]
### Related problem

I'm trying out `nu` and I started using it in `0.61.0`. `0.62.0` came out and changed several things in how the config is supposed to be written. There was no warning, no indication, nothing about it when launching `nu`.

What's more, since the default config is written entirely in `config.nu`, if it evolves upstream it won't be updated for me, leading to potential confusion very quickly when i don't experience the changes noted in the changelog.

Finally, I think there is no way to display/query the full configuration that is active while running `nu` and it makes for a poor UX since I can't discover settings I would like to modify and such.

### Describe the solution you'd like

A `nu config` subcommand that would:

- output a normal table when given no arguments or `--table`
- output a runnable `nu` script when given `--script` so that it can be tried out in isolation

### Do nothing

Possible, still horrible UX. I don't want to have to follow `nu` updates very closely just to update the *default* config.

### Give a warning like `starship` does on update

Better, but starship configs are quite stable and no runnable scripts but a single `toml` file. `nu`'s default configuration changes often, and in non-backward compatible ways.

### Rewrite the `config/env.nu` files on update

This would delete all custom configurations, obviously not feasible. Renaming the old file and sourcing it in the new one is problematic too, since if sourced at the start the new defaults could override configuration changed by the user and if sourced at the end the old defaults could override the new ones, which is not better.

**Comment by fdncred** [comment]
in version 0.62.0 you can see the active config with `$env.config`.

**Comment by poliorcetics** [comment]
No, with `$env.config` I can only see what I have explicitly set, not the active defaults, which, while useful in some cases, is quite useless here

**Comment by poliorcetics** [comment]
I removed the whole `let-env config = ...`:
```
❯ echo $env.config
Error: nu::shell::name_not_found (link)

  × Name not found
   ╭─[entry #1:1:1]
 1 │ echo $env.config
   ·           ───┬──
   ·              ╰── did you mean 'LANG'?
   ╰────
```

**Comment by fdncred** [comment]
Right, `$env.config` is supposed to show you what you have intentionally configured. It's not supposed to show you the default configuration. That is supposed to be the intent of the `default_config.nu` and `default_env.nu` in the repo. Both those files are built into the binary so we should be able to print them with some code changes.

**Comment by Kangaxx-0** [comment]
`$env.config` should not be working since config had been moved into `config.nu`. If someone can elaborate a bit on what needs to be done, I can take a look

**Comment by fdncred** [comment]
If you look at the default_config.nu and the default_env.nu you'll see how to properly setup the nushell environment. We recently changed from `let config = {}` to `let-env config = {}` so while you can set `let config` equal to anything, nushell cannot use it because nushell is looking for an environment variable called config.

**Comment by Kangaxx-0** [comment]
> If you look at the default_config.nu and the default_env.nu you'll see how to properly setup the nushell environment. We recently changed from `let config = {}` to `let-env config = {}` so while you can set `let config` equal to anything, nushell cannot use it because nushell is looking for an environment variable called config.

Yep, I checked the code, all default configs have been built in to binary, and `$env.confg` only picks up the settings from `config.nu`.  What would be the expected behavior, tho? Output the rest of config settings along with `$env` or we want to approach this differently?

**Comment by Kangaxx-0** [comment]
> I don't think $env.config is supposed to show you everything in your config and env files. It shows what's in the config section only. 

Then what is the better way to address this ticket?

**Comment by fdncred** [comment]
Maybe some command that lists the default configurations kind of like we have `keybindings default`? I'm not really sure exactly.

**Comment by poliorcetics** [comment]
> Maybe some command that lists the default configurations kind of like we have keybindings default? I'm not really sure exactly.

That sounds like it is exactly what I wish for. Maybe merge the `keybindings` command in it since it is part of the config ?

To avoid name clashes, `nu config show current` (or `active`), where `nu config` would contain several subcommands, like

- `nu config edit (env|main|dir)` (`env` would open `$nu.env-path` with `$env.EDITOR`, `main` is for `$nu.config-path`, `dir` is `dirname $nu.config-path`)
- `nu config show (<path to dir with files>|current)`, where `current` would show the active config when in a `nu` shell (and an error when not) and `<path to dir with files>` would show what the config would be if `nu` was launched with `<path>/{env.nu,config.nu}`
- `nu config get/set` which is exactly as it looks, update current shell configuration
- `nu config save current` could update the config in `$nu.config-path` to the current one (used in conjunction with the commands above to try out changes and save them once confirmed) 
- `nu config keybindings/keys` same as `keybindings` today (though way longer, is that a problem ? I don't expect keybindings to be used often as a command)

**Comment by fdncred** [comment]
I think most of this sounds good however, i would just name the command `config` with subcommands like `config edit`. I probably wouldn't change or add keybindgings yet. I'm also not sure about the save command. at this point, let's start with the "easy" route of just view and edit that opens an editor of either the config.nu or env.nu. The edit functionality should first check for $env.config.buffer_editor and then check for $env.EDITOR if the prior was not found.

Here's the problem with saving. As i mentioned earlier, $env.config only lists what's in the config environment, NOT what's in the config file itself or the env.nu file itself. For instance, I have a lot of stuff in my config file that is not in the `let-env confg = {}` section. I have comments, I have `let` variables, I have `use` commands, i have `let-env` commands, I have themes. So, if you just `config save` i think it would just overwrite my config file, then i'd be mad because I just lost all my other stuff. Do you see what I'm saying?

**Comment by fdncred** [comment]
The information might exist somewhere in the nu ast and if you can find it and get it out, in the same order it is in within my config files, maintaining comments and all the other mess i rattled off, that would be totally awesome. I'm just thinking that's harder than it sounds. :)

**Comment by poliorcetics** [comment]
I have zero knowledge of `nu`'s internals so I wouldn't know how hard it is.

Anyway, even without the `config save` it would be very nice to have the rest, and since it's not existing anywhere else (I think ?), not having it is not a real downgrade when coming from another shell so it can be postponed without troubles

**Comment by Kangaxx-0** [comment]
Regards to the edit,  nu has got two new commands `config env ` and `config nu`.  For the display work, I need to think about how to bring the information nicely, will probably address `env` as well

<img width="1091" alt="image" src="https://user-images.githubusercontent.com/85712372/169721243-9a136a1b-c7c6-4269-b391-8945e6d85932.png">

**Comment by Kangaxx-0** [comment]
I spent some time to dig through how `$env.config` works(parse, evaluated and finally merged to state), I am thinking, instead of adding another command to show the default configurations, can we update `config.nu` to add the missing default options with some comments.

**Comment by poliorcetics** [comment]
Opened #10480 to scratch my own itch with this issue :D
